mikrotik/resource_interface_wireguard_peer_test.go

```diff
@@ -0,0 +1,189 @@
+		ProtoV5ProviderFactories: testAccProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckMikrotikInterfaceWireguardPeerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccInterfaceWireguardPeer(interfaceName, publicKey),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccInterfaceWireguardPeerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "interface", interfaceName),
+					resource.TestCheckResourceAttr(resourceName, "public_key", publicKey),
+					resource.TestCheckResourceAttr(resourceName, "comment", origCommentPeer)),
+			},
+			{
+				Config: testAccInterfaceWireguardPeerUpdatedComment(interfaceName, publicKey),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccInterfaceWireguardPeerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "interface", interfaceName),
+					resource.TestCheckResourceAttr(resourceName, "comment", updatedCommentPeer)),
+			},
+		},
+	})
+}
+
+func TestAccMikrotikInterfaceWireguardPeer_import(t *testing.T) {
+	client.SkipInterfaceWireguardIfUnsupported(t)
+
+	interfaceName := "tf-acc-interface-wireguard-import"
+	publicKey := "/zYaGiYbgNsSy8AIcxuEcwYwOVdqJJRKG91s9ypwfiM="
+	resourceName := "mikrotik_interface_wireguard_peer.bar"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV5ProviderFactories: testAccProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckMikrotikInterfaceWireguardPeerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccInterfaceWireguardPeer(interfaceName, publicKey),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccInterfaceWireguardPeerExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "interface"),
+					resource.TestCheckResourceAttr(resourceName, "public_key", publicKey),
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+				),
+			},
+			{
+				ResourceName: resourceName,
+				ImportState:  true,
+				ImportStateIdFunc: func(s *terraform.State) (string, error) {
+					return interfaceName, nil
+				},
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccInterfaceWireguardPeer(interfaceName string, publicKey string) string {
+	return fmt.Sprintf(`
+	resource "mikrotik_interface_wireguard" "bar" {
+		name = "%s"
+		comment = "test interface"
+		listen_port = "12321"
+		mtu = "1420"
+	}
+	resource "mikrotik_interface_wireguard_peer" "bar" {
+		interface = mikrotik_interface_wireguard.bar.name
+		public_key = "%s"
+		comment = "%s"
+		allowed_address = "%s"
+		endpoint_port = "%d"
+	}
+	`, interfaceName, publicKey, origCommentPeer, origAllowedAddress, origEndpointPort)
+}
+
+func testAccInterfaceWireguardPeerUpdatedComment(interfaceName string, publicKey string) string {
+	return fmt.Sprintf(`
+	resource "mikrotik_interface_wireguard" "bar" {
+		name = "%s"
+		comment = "test interface"
+		listen_port = "12321"
+		mtu = "1420"
+	}
```

Comment:
Rather than duplicating a `mikrotik_interface_wireguard` resource string multiple times, we should reuse the one from [this test file](https://github.com/ddelnano/terraform-provider-mikrotik/blob/fdce5dad4079b7076fdf51fb6d97e2250876980b/mikrotik/resource_interface_wireguard_test.go#L101). You will need to parameterize that function to make it work. Below is what the code I'm thinking of will look like:

```go
func testAccInterfaceWireguardPeerUpdatedComment(interfaceName string, publicKey string) string {
	return testAccInterfaceWireguard(interfaceName) + fmt.Sprintf(`

	resource "mikrotik_interface_wireguard_peer" "bar" {
		interface = mikrotik_interface_wireguard.bar.name
		public_key = "%s"
		comment = "%s"
		allowed_address = "%s"
		endpoint_port = "%d"
	}
	`,  publicKey, updatedCommentPeer, origAllowedAddress, origEndpointPort)
}
```